sympy/functions/elementary/trigonometric.py

```diff
@@ -1047,6 +1099,31 @@ def eval(cls, arg):
                 return None
 
             if pi_coeff.is_Rational:
+                if pi_coeff.q > 2 and not pi_coeff.q % 2:
+                    narg = pi_coeff*S.Pi*2
+                    cresult, sresult = cos(narg), cos(narg - S.Pi/2)
+                    if not isinstance(cresult, cos) \
+                            and not isinstance(sresult, cos):
+                        if sresult == 0:
+                            return S.ComplexInfinity
```

Comment:
I guess that line never gets called, since if q > 2 then 2/q < 1 and sin(2*pi/q) != 0.  So I can get rid of the lines:
```python
if sresult == 0:
    return S.ComplexInfinity
``` 

I will also squash the commits